aion/web/elm/src/Update.elm

```diff
@@ -121,32 +140,32 @@ update msg model =
             else
                 model ! []
 
-        NoOperation ->
-            model ! []
-
-        SetNewQuestionContent questionContent ->
+        UpdateCreateQuestionForm name value ->
             let
                 oldPanelData =
                     model.panelData
             in
-                { model | panelData = { oldPanelData | newQuestionContent = questionContent } } ! []
+                { model
+                    | panelData =
+                        { oldPanelData | createQuestionForm = Forms.updateFormInput oldPanelData.createQuestionForm name value }
+                }
+                    ! []
 
-        SetNewAnswerContent answerContent ->
+        CreateNewQuestionWithAnswers ->
             let
-                oldPanelData =
-                    model.panelData
+                possibleFields =
+                    [ "question", "answers", "subject" ]
+
+                validationErrors =
+                    possibleFields
+                        |> List.map (\name -> Forms.errorList model.panelData.createQuestionForm name)
+                        |> List.foldr (++) []
+                        |> List.filter (\validations -> validations /= Nothing)
             in
-                { model | panelData = { oldPanelData | newAnswerContent = answerContent } } ! []
-
-        SetNewAnswerCategory answerCategoryName ->
-            let
-                answerCategoryToId =
-                    getSubjectIdByName model.rooms answerCategoryName
-
-                oldPanelData =
-                    model.panelData
-            in
-                { model | panelData = { oldPanelData | newAnswerCategory = answerCategoryToId } } ! []
+                if List.isEmpty validationErrors then
+                    ( model, createQuestionWithAnswers model )
```

Comment:
I think it's a matter or #56 but I wouldn't pass the whole model to any function (apart from main **Update** etc). I believe that we should always only pass one data structure from our model. `Well, what if I would then need to pass like three different structures from our model, eh?` one could ask. I believe that if we split our state well, these all three different structures should end up in one bigger structure that aggregates them.